buildvertical.targets

```diff
@@ -51,35 +58,57 @@
       <_BestConfiguration Condition="'$(BuildConfigurations)' == ''">$(BuildConfiguration)</_BestConfiguration>
     </PropertyGroup>
   </Target>
-
-  <!-- Runs in a leaf project (eg: csproj) to determine best configuration if one exists for purposes of project references -->
-  <Target Name="FindBestConfigurationForProjectReference"
-          DependsOnTargets="FindBestConfiguration"
-          Returns="$(_BestConfiguration)">
-    <Error Condition="'$(_BestConfiguration)' == ''" Text="Could not find a configuration for $(BuildConfiguration) from $(BuildConfigurations)" />
+  
+  <!-- Runs in a leaf project (eg: csproj) to determine all configurations -->
+  <Target Name="GetBuildConfigurations"
+          Returns="$(_AllBuildConfigurations)">
+     <PropertyGroup>
+       <_AllBuildConfigurations>$(BuildConfigurations)</_AllBuildConfigurations>
+       <_AllBuildConfigurations Condition="'$(BuildConfigurations)' == ''">$(BuildConfiguration)</_AllBuildConfigurations>
+     </PropertyGroup>
   </Target>
 
   <!-- Runs in a leaf project (csproj) to determine best configuration for ProjectReferences -->
   <Target Name="AnnotateProjectReference"
-          BeforeTargets="BeforeResolveReferences">
+          BeforeTargets="BeforeResolveReferences"
+          Condition="'@(ProjectReference)' != ''"
+          Inputs="%(ProjectReference.Identity)"
+          Outputs="unused">
 
-    <!-- find the best configuration for each project, projects that do not have an 
-        applicable configuration will fail the build.  -->
-    <MSBuild Targets="FindBestConfigurationForProjectReference"
+    <!-- Get all configurations for each ProjectReference.  We use this 
+         rather than the FindBestConfiguration target since we need to 
+         choose the best configuration for the referencing project's 
+         configuration, not the BuildConfiguration.  -->
+    <MSBuild Targets="GetBuildConfigurations"
              Projects="@(ProjectReference)">
-      <Output TaskParameter="TargetOutputs" ItemName="_ProjectReferenceBestConfigurations" />
+      <Output TaskParameter="TargetOutputs" PropertyName="_projectReferenceBuildConfigurations" />
     </MSBuild>
 
+    <!-- Find the best configuration for the current Project's Configuration -->
+    <FindBestConfiguration Condition="'$(_projectReferenceConfiguration)' == ''"
```

Comment:
Good catch, previously I allowed this to be specified as metadata on the project reference, this was the property that pulled it out.  I removed that support because it was most often the wrong thing to do, but can add it back if you think it might provide value.